frontend/app/handlers.ts

```diff
@@ -0,0 +1,21 @@
+// Make API call to server to check if password was found in breached dataset
+export const checkSecurity = async () => {
+  const response = await fetch("http://localhost:18080", {
+    method: "GET",
+    mode: "cors",
+    headers: {
+      "Access-Control-Allow-Headers": "*",
+    },
+  })
+    .then((res) => {
+      return res.json();
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+      return {
+        status: "error",
+      };
+    });
+
+  return response;
```

Comment:
**issue**: When working with promises in TypeScript/JavaScript, you usually use only `async`/`await` _or_ `.then()` etc. My understanding is that `async`/`await` is just syntactic sugar for the promise resolution you are using. So, you can just do something like this:
```
const checkSecurity = async ()=>{
    try {
    const response = await fetch("http://localhost:18080", {
    method: "GET",
    mode: "cors",
    headers: {
      "Access-Control-Allow-Headers": "*",
    },
  })
        const data = await response.json();
        return data
    } catch(error){
        console.log("Error fetching data ",error);
        return {status: "error"};
    }  
}
```
(Formatting needs work). Please correct me if I'm wrong.